paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentElementLoader.kt

```diff
@@ -531,10 +531,10 @@ internal class DefaultPaymentElementLoader @Inject constructor(
         elementsSession: ElementsSession,
         initializationMode: PaymentElementLoader.InitializationMode
     ): LinkConfiguration? {
-        if (!configuration.link.shouldDisplay ||
-            configuration.billingDetailsCollectionConfiguration.collectsAnything ||
-            !elementsSession.isLinkEnabled
-        ) {
+        val collectsUnsupportedBillingDetails = with(configuration.billingDetailsCollectionConfiguration) {
+            collectsEmail || collectsName
+        }
```

Comment:
`CardEditUI` does not seem to support email / name / phone collection (@toluo-stripe am I right?) so we keep disabling Link for now in these scenarios.

See iOS update flow when any of these is `Always`:

<img width="311" alt="image" src="https://github.com/user-attachments/assets/22182ec0-22a6-4689-b8a8-1dcd81044cd8" />